client/src/services/users-service.ts

```diff
@@ -0,0 +1,28 @@
+import { User } from '../types/user';
+
+export async function getUsers() {
+  return fetch('/api/users')
+    .then((res) => res.json())
+    .then((res) => res.users);
+}
+
+export async function postUser(formData: FormData): Promise<User> {
+  const formObject = Object.fromEntries(formData.entries());
+  const body = JSON.stringify(formObject);
+  const options = {
+    method: 'POST',
+    body,
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+  };
+
+  const response = await fetch('/api/users', options);
+
+  if (response.ok === false) {
+    throw new Error(await response.text());
```

Comment:
When the response is not in 2xx family its body will contain JSON with Problem object. So using `.text()` here is kind of useless. In my opinion we should propagate the Problem object up to the view layer so that we could render the error message to the user. Sad that we cannot specify the rejection type of a Promise.

TLDR
```suggestion
    throw new Error(await response.json());
```